server/src/main/java/com/coincoinche/engine/CoincheGame.java

```diff
@@ -0,0 +1,66 @@
+package com.coincoinche.engine;
+
+import com.coincoinche.engine.game.GameResult;
+import com.coincoinche.engine.game.RedBlueRotatingPlayersGame;
+import com.coincoinche.engine.teams.Player;
+import com.coincoinche.engine.teams.Team;
+
+/** Implementation of a coinche game. */
+public class CoincheGame extends RedBlueRotatingPlayersGame<Player> {
+
+  private static final int WINNING_POINTS = 1000;
+  // TODO nockty: consider DI for this
+  private CoincheGameRound currentRound;
+
+  public CoincheGame(Team redTeam, Team blueTeam) {
+    super(redTeam, blueTeam);
+    initializeGameRound();
+  }
+
+  GameResult<Team> moveWasApplied() {
+    GameResult<Team> result = currentRound.moveWasApplied();
+    if (!result.isFinished()) {
+      return GameResult.unfinishedResult();
+    }
+    result.getWinnerTeam().addPoints(result.getWinnerPoints());
+    Team winnerTeam = getWinnerTeam();
+    if (winnerTeam != null) {
+      return GameResult.finishedResult(winnerTeam, winnerTeam.getPoints());
+    }
+    updateGameRound();
+    return GameResult.unfinishedResult();
+  }
+
+  private Team getWinnerTeam() {
```

Comment:
I assumed it couldn't happen (depends on the rules we choose). But that's a good point, it's better if we handle this case now so that it works even if we change rules in the future.

Reply:
Even with the current rules, I don't see how it can't happen :p

Reply:
If only one team earns points in a given round, I think it shouldn't happen.